visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -25,28 +25,26 @@ export class CodeChartaService {
 
 	async loadFiles(nameDataPairs: NameDataPair[]) {
 		this.fileStates = this.storeService.getState().files
+		const errors = []
+		const warnings = []
+
 		for (const nameDataPair of nameDataPairs) {
-			try {
-				validate(nameDataPair)
+			errors.push(...checkErrors(nameDataPair.content))
+			if (errors.length > 0) {
+				this.fileStates.filter(file => getCCFile(nameDataPair) !== file.file)
+				this.recentFiles.filter(fileName => fileName !== nameDataPair.fileName)
+			}
+			if (errors.length === 0) {
```

Comment:
The behavior before was different. Errors and warnings were shown at the same time. Now, Warnings are shown only if no errors are present.

Reply:
That is true. But it didn't make sense to me. Let's say a file has an error, which means something is seriously wrong with it. Why should we bother checking for warnings at this point? Plus, in the future if we decide to add warnings that assume that the file structure is correct, the old implementation is not flexible enough.
When we show the user the validation results, I think if a file has an error, the user won't care about its warnings. But once the error is fixed, the warnings will be checked for and shown if found.

Basically, the warnings of a certain file will be shown only when it has no errors, meaning the structure is correct. 
What do you think?